### PR TITLE
fix(nullattributes): Accepting null attributes and eventTags

### PR DESCRIPTION
**Pull request by msohailhussain**
NULL support for attributes and eventTags.

### CODE REVIEW

**Comment by mikeproeng37** [comment]
build

File: OptimizelySDKCore/OptimizelySDKCore/OPTLYDecisionService.m [hunk 1]
```diff
@@ -164,7 +164,7 @@ - (NSString *)getBucketingId:(NSString *)userId
     // If the bucketing ID key is defined in attributes, then use that
     // in place of the userID for the murmur hash key
     if (![OPTLYDecisionService isEmptyString:attributes[OptimizelyBucketId]]) {
```
Comment (thomaszurkan-optimizely) [inline OptimizelySDKCore/OptimizelySDKCore/OPTLYDecisionService.m]: how about if we renamed that to getValidString where it returns the string as a string if it is valid or null?  Then, you wouldn't have that weird cast (that gives a warning in xcode).  

String validBucketingId = [OPTLYDecisionService isEmptyString:attributes[OptimizelyBucketId]];
if (validBucketingId != null)  {
  bucketingId = validBucketingId

**Review by thomaszurkan-optimizely — CHANGES_REQUESTED** [review]
Can we either rename the isEmptyString or slightly refactor to make that more obvious that it tests type as well as whether the string was empty?

File: OptimizelySDKCore/OptimizelySDKCore/OPTLYDecisionService.m (hunk 1 above)
Comment (yasirfolio3) [reply OptimizelySDKCore/OptimizelySDKCore/OPTLYDecisionService.m]: Please review.

**Review by thomaszurkan-optimizely — APPROVED** [review]
@msohailhussain thanks for the changes!  Looks good!

**Comment by thomaszurkan-optimizely** [comment]
@msohailhussain can you close this pr and open it against master?  thanks

File: OptimizelySDKCore/OptimizelySDKCore/OPTLYDecisionService.m [hunk 2]
```diff
@@ -163,8 +164,9 @@ - (NSString *)getBucketingId:(NSString *)userId
     NSString *bucketingId = userId;
     // If the bucketing ID key is defined in attributes, then use that
     // in place of the userID for the murmur hash key
-    if (![OPTLYDecisionService isEmptyString:attributes[OptimizelyBucketId]]) {
-        bucketingId = (NSString *)attributes[OptimizelyBucketId];
+    
+    for (NSString *validBucketingId = [attributes[OptimizelyBucketId] getValidString]; validBucketingId != nil; validBucketingId = nil) {
```
Comment (msohailhussain) [inline OptimizelySDKCore/OptimizelySDKCore/OPTLYDecisionService.m:168]: @yasirfolio3 Why for loop is used instead of if.

**Comment by msohailhussain** [comment]
@thomaszurkan-optimizely Closing this PR. And created new PR. 
https://github.com/optimizely/objective-c-sdk/pull/335

File: OptimizelySDKCore/OptimizelySDKCore/OPTLYDecisionService.m (hunk 2 above)
Comment (yasirfolio3) [reply OptimizelySDKCore/OptimizelySDKCore/OPTLYDecisionService.m:168]: in case of if getValidString != nil, the getValidString method would be called twice, the above implemented way is just an alternate way of using "if let" from swift where getValidString would only be required to call once, attached is the reference:

http://blog.cristik.com/2016/01/if-let-statement-in-objective-c/
